<html>
	
<head>
<title>PCPhaser</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<link href="/css/normal.css" rel="stylesheet" type="text/css">
</head>

<body>
<h1>Users' documentation for PCPhaser</h1>
<p>PCPhaser <a href="LICENSE.txt">(copyright 2013 Universidad de Granada)</a> is an application which allows you to use the <a href="http://www.mybiosoftware.com/population-genetics/1309">EIGENSOFT</a> package to obtain PCA graphs from the individual's haplotypes, rather than from their genotypes.</p>

<h2>System requirements</h2>
<p>PCPhaser can be run only in Linux architectures, and you will need to install the following packages:</p>

<ul>
<li>gcc 4.6 or newer, in order to run EIGENSOFT.</li>
	<ul>		
	<li>For Debian-based distributions: sudo apt-get install build-essential</li>
	<li>For Fedora / CentOS: yum install gcc</li>
	</ul>
<li>moreutils package, in order to use the sponge tool.</li>
	<ul>
	<li>For Debian-based distributions: apt-get install moreutils</li>
	<li>For Fedora / CentOS: yum install moreutils</li>
	</ul>
<li>lapack package, in order to compute the eigenvectors.</li>
	<ul>
	<li>For Debian-based distributions: apt-get install liblapack-dev</li>
	<li>For Fedora / CentOS: yum install lapack</li>
	</ul>
<li>gnuplot package, in order to draw the graphs.</li>
	<ul>
	<li>For Debian-based distributions: apt-get install gnuplot</li>
	<li>For Fedora / CentOS: yum install gnuplot</li>
	</ul>
</ul>

<p>You will not need to install EIGENSOFT, since a copy of the package -version 4.2- is included in the application folder. However, you must not move the script "PCPhaser.sh", or it will not work properly, since it uses relative paths to other scripts in the application folder.</p>

<h2>Download PCPhaser</h2>
<p>PCPhaser 1.2 can be downloaded <a href="PCPhaser.v1.2.zip">here</a>.</p>

<h3>Install PCPhaser</h3>
<p>Once file PCPhaser.zip has been downladed, your should execute this commands in console to install it:</p>

<ol>
<li><code>cd [download folder]</code> <i>(to open the folder that contains the ".zip" file)</i></li>
<li><code>unzip PCPhaser.zip</code> <i>(to uncompress the ".zip" file)</i></li>
<li><code>mv -R PCPhaser [target folder]</code> <i>(to move PCPhaser folder to another location)</i></li>
</ol>

<h3>Execute PCPhaser</h3>
<p>PCPhaser requires five arguments, and may be provided with a flag too, as follows:</p>
<code>./PCPhaser.sh [INPUT_FOLDER] [OUTPUT_FOLDER] [OUTPUT_PREFIX] [WIDTH] [HEIGHT] [--silent]</code>

<ul>
<li><code>INPUT_FOLDER</code> is the path containing the ".ped" and the ".map" files with the information about the individuals and the loci. The files can be named in any way, it is only assumed that:
	<ol>
	<li>All SNPs from all the individuals from all the populations that belong to the same chromosome are in the same file.</li>
	<li>Both ".ped" and the ".map" files can contain an integer number just before the extension, indicating the chromosome number.</li>
	<li>There are as many ".map" files as chromosomes considered in the sample, and they can be numbered before the extension too.</li>
	</ol>
<li><code>OUTPUT_FOLDER</code> is the path in which the new files will be created. If the execution ends succesfully, in this folder there will be:
	<ol>
	<li>As many ".ped" files as in INPUT_FOLDER, named the same way, containing the splitted haplotypes.</li>
	<li>An ".evec" file, containing the results of the PCA analysis.</li>
	<li>A ".png" file, representing the PCA graph.</li>
	</ol>
<li><code>OUTPUT_PREFIX</code> is the name will be given to the output ".evec" and ".png" files.</li>
<li><code>WIDTH</code> is the widht, in pixels, for the output PCA graph.</li>
<li><code>HEIGHT</code> is the height, in pixels, for the output PCA graph.</li>
<li><code>--silent</code> is an optional flag that when present, prevents the program to open a window displaying the graph after finishing the execution.</li>
</ul>

<h2>File formats</h2> 
<p>PCPhaser uses the PLINK format for the input and output ".ped" and ".map" files, which is described here:</p>

<ul>
<li><a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped">".ped" format</a></li>
<li><a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#map">".map" format</a></li>
</ul>

<p>The 6th column of ".ped" files -the phenotype- should contain the name of the population where the individual belongs to. As an example, the following ".ped" file has 5 individuals, 3 belonging to the HapMap population CEPH (CEU) and 2 to the HapMap population from Southest USA with African ancestry (ASW):</p>

<code>2427 2427_19909 0 0 2 ASW 2 2 2 2 2 4 1 1 3 1 3 1 2 4 4 2 3 3 1 1 1 3 4 4 2 2 2 2 2 2 1 2 1 1 1 1 1 3 3 4</code><br>
<code>2446 2446_20127 0 0 2 ASW 2 2 2 2 2 2 1 1 3 3 3 3 2 2 4 4 3 3 1 3 1 1 4 4 2 2 1 2 4 2 2 1 3 1 1 1 3 3 4 4</code><br>
<code>1463 1463_12890 0 0 2 CEU 2 2 2 2 2 2 1 1 1 3 1 3 4 2 2 4 3 3 1 1 3 1 4 4 2 2 2 1 2 4 2 2 1 3 1 1 3 3 4 4</code><br>
<code>1463 1463_12889 0 0 1 CEU 2 2 2 2 2 2 1 1 3 3 3 3 2 2 4 4 3 1 3 1 1 1 4 4 2 2 2 2 2 2 1 1 1 1 1 1 3 3 4 4</code><br>
<code>1424 1424_11931 0 0 2 CEU 2 2 2 2 2 2 1 1 3 3 3 3 2 2 4 4 3 3 1 1 1 1 4 4 2 2 2 2 2 2 1 1 1 1 1 1 3 3 4 4</code><br>

<h3>Input files</h3>
<p>For PCPhaser to work, input ".ped" files must contain phased genotypes. Therefore, alleles at every marker are always considered ordered so that alleles in the first position of a genotype belong to the same homologous chromosome. As an example, the second individual in the ".ped" file above (ID# <code>2446_20127</code>) has haplotypes "<code>22213324311421423134</code> / <code>22213324331422211134</code>".</p>

<h3>Output files</h3>
<p>For each input ".ped" file with <b>m</b> lines corresponding to <b>m</b> different individuals, PCPhaser will output another ".ped" file with <b>2m</b> lines, i.e., 2 for each individual, with their haplotypes, but duplicating each allele. As an example, if an individual haplotype is "24322", the haplotype with duplicated alleles will be "2244332222". The only reason to duplicate alleles is technical, as PCPhaser has been implemented as a wrapper of EIGENSOFT. Therefore, lines in the output ".ped" file corresponding to individual 2446_20127 will have the following content:</p>
<code>2446 2446_20127_1 0 0 2 ASW 2 2 2 2 2 2 1 1 3 3 3 3 2 2 4 4 3 3 1 1 1 1 4 4 2 2 1 1 4 4 2 2 3 3 1 1 3 3 4 4</code><br>
<code>2446 2446_20127_2 0 0 2 ASW 2 2 2 2 2 2 1 1 3 3 3 3 2 2 4 4 3 3 3 3 1 1 4 4 2 2 2 2 2 2 1 1 1 1 1 1 3 3 4 4</code><br>

<h2>Use example</h2>
<p>You can test the application using the example data included in the application, by running the following command:</p>
<code>./PCPhaser.sh exampleIN/ exampleOUT/ ASW+CEU 1000 800</code>

<p>When the proccess is completed, a window showing the PCA graph will pop out by default. If you are going to work with huge data files, it is strongly recommended to run the program in the background and in silent mode:</p>
<code>nohup ./PCPhaser.sh exampleIN/ exampleOUT/ ASW+CEU 1000 800 --silent &</code>

<p>The plot drawn by PCPhaser from this example can be seen at <a href="http://bios.ugr.es/PCPhaser/ASW+CEU.png">http://bios.ugr.es/PCPhaser/ASW+CEU.png</a>.</p>

<!--<h2>Citation</h2>
<p>Please, cite PCPhaser with the following paper:
</p><p>
Abad-Grau, M.M.; Montes, R.; Sebastiani, P. <br>&quot;Building chromosome-wide LD maps&quot;, <br>Bioinformatics 22(16): 1933-1934 (2006)</p>
-->

<!--<h3>Additional information</h3>

<a href="supplementary/data.html">Some datasets</a> 
<br> <a href="supplementary">Other supplementary information</a> 
-->
</body>
</html>
